server/routes/bookings/bookingDetailsController.test.ts

```diff
@@ -0,0 +1,70 @@
+import type { Express } from 'express'
+import request from 'supertest'
+import * as cheerio from 'cheerio'
+import { SessionData } from 'express-session'
+import { appWithAllRoutes } from '../testutils/appSetup'
+import { createMockBookerService, createMockVisitService } from '../../services/testutils/mocks'
+import TestData from '../testutils/testData'
+import paths from '../../constants/paths'
+
+let app: Express
+
+const bookerService = createMockBookerService()
+const visitService = createMockVisitService()
+const visit = TestData.visitDto()
+const bookerReference = TestData.bookerReference().value
+const prisoner = TestData.prisoner({ prisonCode: 'DHI' })
+let sessionData: SessionData
+
+beforeEach(() => {
+  sessionData = {
+    booker: {
+      reference: bookerReference,
+      prisoners: [prisoner],
+    },
+    bookings: [visit],
+  } as SessionData
+  visitService.getFuturePublicVisits.mockResolvedValue([visit])
+  app = appWithAllRoutes({ services: { bookerService, visitService }, sessionData })
+})
+
+afterEach(() => {
+  jest.resetAllMocks()
+})
+
+describe('View single booking', () => {
+  it('should render the booking details page', () => {
+    return request(app)
+      .get(`${paths.BOOKINGS.VISIT}/1`)
+      .expect('Content-Type', /html/)
+      .expect(res => {
+        const $ = cheerio.load(res.text)
+        expect($('title').text()).toMatch(/^Bookings -/)
+        expect($('[data-test="back-link"]').length).toBe(1)
```

Comment:
```suggestion
        expect($('[data-test="back-link"]').attr('href')).toBe(paths.BOOKINGS.HOME)
```
Should check that the link is correct; not just that it exists.